Octokit.Tests.Integration/GithubClientExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Octokit.Tests.Integration
+{
+    public static class GithubClientExtensions
+    {
+        public async static Task<DisposableRepository> CreateDisposableRepository(this IGitHubClient client, NewRepository newRepository)
+        {
+            return await ((TestRepositoriesClient)client.Repository).CreateDisposableRepository(newRepository);
+        }
+    }
+
+    public class DisposableRepository : Repository, IDisposable
```

Comment:
I'd love to not need this class, but instead have tests access the repository directly, e.g:

```
public class DisposableRepository : IDisposable
{
    public Repository Repository { get; private set; }
    public static DisposableRepository Init(Repository repository)
    { 
         var disp = new DisposableRepository() { Repository = repository };
         return disp;
    }
    /* IDisposable code here */
}
```

But that means we'd need to change the tests to read that inner property. I'm somewhat neutral on which approach is better, but the reflection code in here concerns me a bit.